### PR TITLE
Provide integration tests for the key functionalities

### PR DESCRIPTION
**Pull request by antonia-avramova**
[#19] Provide integration tests for the key functionalities

Signed-off-by: Antonia Avramova <antonia.avramova@bosch.io>

### CODE REVIEW

File: integration/bootstrapping_test.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	DeviceRegistryAPIUsername string `env:"DEVICE_REGISTRY_API_USERNAME" envDefault:"ditto"`
+	DeviceRegistryAPIPassword string `env:"DEVICE_REGISTRY_API_PASSWORD" envDefault:"ditto"`
+}
+
+type bootstrappingSuite struct {
+	suite.Suite
+
+	util.SuiteInitializer
+
+	bootstrappingThingURL   string
```
Comment (antoniyatrifonova) [inline integration/bootstrapping_test.go:64]: You don't need it, remove it

File: integration/bootstrapping_test.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+		mqttClient.Disconnect(uint(cfg.MQTTQuiesceMS))
+		require.NoError(t, err, "initialize ditto client")
+	}
+
+	suite.Cfg = cfg
+	suite.DittoClient = dittoClient
+	suite.MQTTClient = mqttClient
+	suite.ThingCfg, suite.requestID, err = getThingConfigurationBootstrapping(t, cfg)
+	if err != nil {
+		defer suite.TearDown()
```
Comment (antoniyatrifonova) [inline integration/bootstrapping_test.go:98]: You don't need the defer. Please, review the use of defer in all places

**Review by antoniyatrifonova — CHANGES_REQUESTED** [review]

**Review by antoniyatrifonova — APPROVED** [review]

File: integration/bootstrapping_test.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	bootstrappingFeatureID             = "Bootstrapping"
+	actionResponse                     = "response"
+	msgFailedCreateWebsocketConnection = "failed to create websocket connection"
+	eventFilterTemplate                = "like(resource:path,'/features/%s/*')"
+	sbServiceName                      = "suite-bootstrapping.service"
+	suiteConnector                     = "suite-connector.service"
```
Comment (k-gostev) [inline integration/bootstrapping_test.go:42]: align it with the constant above, eg. `scServiceName` or `suiteConnectorServiceName`

File: integration/bootstrapping_test.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+		require.NoError(t, err, "initialize ditto client")
+	}
+
+	suite.Cfg = cfg
+	suite.DittoClient = dittoClient
+	suite.MQTTClient = mqttClient
+	suite.ThingCfg, suite.requestID, err = getThingConfigurationBootstrapping(t, cfg)
+	if err != nil {
+		require.NoError(t, err, "cannot get thing configuration")
+	}
```
Comment (k-gostev) [inline integration/bootstrapping_test.go:103]:
```suggestion
		require.NoError(t, err, "cannot get thing configuration")
```

File: integration/util.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/eclipse-kanto/kanto/integration/util"
+)
+
+func getBootstrapConfigStruct(path string) (util.BootstrapConfiguration, error) {
```
Comment (k-gostev) [inline integration/util.go:27]: you can use pointer here in order to return `nil, err` on error.
```suggestion
func getBootstrapConfigStruct(path string) (*util.BootstrapConfiguration, error) {
```

File: integration/util.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+	cfg := util.BootstrapConfiguration{}
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return cfg, fmt.Errorf("unable to read file '%s': %v", path, err)
+	}
+	err = json.Unmarshal(content, &cfg)
+	if err != nil {
+		return cfg, fmt.Errorf("unable to unmarshal to json: %v", err)
+	}
```
Comment (k-gostev) [inline integration/util.go:36]:
```suggestion	
	if err := json.Unmarshal(content, &cfg); err != nil {
		return cfg, fmt.Errorf("unable to unmarshal to json: %v", err)
	}
```

File: integration/util.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+func parseString(value interface{}) (string, error) {
+	property, check := value.(string)
+	if !check {
+		return "", fmt.Errorf("failed to parse the property to string")
+	}
+	return property, nil
+}
+
+func createBootstrappingResponsePayload(
+	cfg *util.ConnectorConfiguration, requestID string) (map[string]interface{}, error) {
```
Comment (k-gostev) [inline integration/util.go:57]:
```suggestion
func createBootstrappingResponsePayload(cfg *util.ConnectorConfiguration, requestID string) (map[string]interface{}, error) {
```

**Review by k-gostev — CHANGES_REQUESTED** [review]
couple of nits

**Review by k-gostev — APPROVED** [review]

**Review by antoniyatrifonova — APPROVED** [review]

**Pull request merged**